virttest/utils_iptables.py

```diff
@@ -84,3 +87,215 @@ def setup_or_cleanup_iptables_rules(cls, rules, params=None,
         # cleanup server session
         if params:
             server_session.close()
+
+
+def singleton_class(class_name):
```

Comment:
Maybe we need take more consideration for this singleton class. Current key issue (requirement) is that we need two firewalld objects, one is for local host and the other is for remote host in migration. But this implementation can not support this as it will always return the firewalld object created for local in preprocess for your dhcp fix.

So we are thinking if you could use two classes, one fore local firewalld, the other for remote firewalld, and both can be implemented with singleton. Or you could use other ways you like to support the requirement.  Thanks.

Reply:
Two classes are not required, I have fixed it in decorator. Please help to check it in your environment.

Reply:
lets have this method outside, may be in a separate class/file, so that others will also get benefit..thanks